src/sentry/search/snuba/backend.py

```diff
@@ -262,7 +267,8 @@ def _query(self, project, retention_window_start, group_queryset, tags, environm
 
 
 def do_search(project_id, environment_id, tags, start, end,
-              sort, candidates=None, limit=1000, **parameters):
+              sort, extra_aggregations, candidates=None, limit=1000, **parameters):
```

Comment:
I wasn't thinking about it before but if `limit` is less than the number of total candidate hashes isn't this going to return partial results for groups (even more likely if no explicit candidates are provided?)

Reply:
> less than the number of total candidate hashes

True, I guess we want `min(limit, len(hashes))` right?

> even more likely if no explicit candidates are provided

Not sure what else to do in this case. I don't think we should return all of the project's hashes? So 1000 is arbitrary, for sure, but I feel like we need to pick something unless we find a way to mirror all group data in to CH?

Reply:
I don't really have a good answer to this but I think we need to seriously consider if this is acceptable behavior for using it to serve production requests (if it's not obvious, my opinion is that it isn't.)

Reply:
I totally agree, going to work on that outside of this PR: https://getsentry.atlassian.net/browse/SNS-58